java/server/src/org/openqa/grid/internal/RemoteProxy.java

```diff
@@ -39,6 +39,16 @@
  * only support Firefox.
  */
 public interface RemoteProxy extends Comparable<RemoteProxy> {
+
+  /**
+   *
+   * @param protocol - A {@link SeleniumProtocol} object that identifies the request flavor.
+   * @param capabilities - the type of test the client is interested in performing.
+   * @return - The entity on a proxy that can host a test session.
+   */
+  default TestSlot newTestSlot(SeleniumProtocol protocol, Map<String, Object> capabilities) {
```

Comment:
ping @mach6 . Any updates on this ?

ping @lukeis - If there are no updates from @mach6 , please let me know what more is needed from my side to get this merged.